parent/pom.xml

```diff
@@ -1482,5 +1491,66 @@
         </plugins>
       </build>
     </profile>
+
+    <!-- profile to disable the fmt-plugin on older jdks -->
+    <profile>
+      <id>fmt-plugin</id>
+      <activation>
+        <jdk>[,9)</jdk>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <artifactId>fmt-maven-plugin</artifactId>
+            <groupId>com.coveo</groupId>
+            <executions>
+              <execution>
+                <id>format</id>
+                <phase>none</phase>
+                <goals>
+                  <goal>check</goal>
+                  <goal>format</goal>
+                </goals>
+              </execution>
+            </executions>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+
+    <!-- profile to perform validation checks -->
```

Comment:
Nit: can you explain why checkstyle is not a validation check? Here I'm mostly confused by the comment, not by the profile, if that makes sense.

Reply:
The checkstyle validation, the license format and the googlestyle (fmt) format are all executed by default (i.e. without activating a profile). These format steps make sense to execute locally (by default), but in the ci pipeline you don't want to make changes to the code anymore, so the stricter `check` is used instead with this new `checks` profile. 

Now, if the checkstyle plugin would offer a quickfix/format goal, it would make sense to move the checkstyle check goal into the `checks` profile. Sadly, it doesn't. 

So to me, the difference is that these checks are stricter than the default execution.

To reflect this, I could change the comment to something like:

```suggestion
    <!-- profile to perform strict validation checks -->
```

or we could rename it altogether to strict-checks or so. WDYT?

Reply:
Sounds good :+1: